tests/lib/retry.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+snap_install() {
+    local PARAMS=$@
+    local CONDITIONS='timeout \| connection refused'
+    local RETRIES=2
+    local SLEEP=2
+
+    local ITER=0
+    until [ $ITER -gt $RETRIES ]; do
+        ERROR=$(snap install $PARAMS 2>&1> /dev/null)
+        if [ -z "$ERROR" ]; then 
+            break
+        elif [[ $( echo $ERROR | grep -c "$CONDITIONS") -ge 1 ]]; then
```

Comment:
Instead of `grep` we are using the `MATCH` function defined by spread, beside other benefits it makes diagnosing much easier.